<?xml version="1.0"?>
<!--
Copyright (c) 2013, Goncalo Cabrita/ISR University of Coimbra
Copyright (c) 2012, Ryan Gariepy/Clearpath Robotics
All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:
    * Redistributions of source code must retain the above copyright
      notice, this list of conditions and the following disclaimer.
    * Redistributions in binary form must reproduce the above copyright
      notice, this list of conditions and the following disclaimer in the
      documentation and/or other materials provided with the distribution.
    * Neither the name of Clearpath Robotics nor the
      names of its contributors may be used to endorse or promote products
      derived from this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS BE LIABLE FOR ANY
DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

This program contains the description of the HRATC2017 Husky robot based on the original Clearpath 
Husky A200 file and modified by the ISR University of Coimbra
-->

<robot 

    xmlns:xi="http://www.w3.org/2001/XInclude"
    xmlns:gazebo="http://playerstage.sourceforge.net/gazebo/xmlschema/#gz"
    xmlns:model="http://playerstage.sourceforge.net/gazebo/xmlschema/#model"
    xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor"
    xmlns:body="http://playerstage.sourceforge.net/gazebo/xmlschema/#body"
    xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom"
    xmlns:joint="http://playerstage.sourceforge.net/gazebo/xmlschema/#joint"
    xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
    xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface"
    xmlns:rendering="http://playerstage.sourceforge.net/gazebo/xmlschema/#rendering"
    xmlns:renderable="http://playerstage.sourceforge.net/gazebo/xmlschema/#renderable"
    xmlns:physics="http://playerstage.sourceforge.net/gazebo/xmlschema/#physics"
    xmlns:xacro="http://ros.org/wiki/xacro" name="husky">
       
    <!-- Included URDF/XACRO Files -->
    <xacro:include filename="$(find hratc2017_framework)/urdf/fsr_husky_parts/gps_antenna.urdf.xacro" />

    <link name="base_footprint">
        <!--<inertial>
            <mass value="0.0001" />
            <origin xyz="0 0 0" />
            <inertia ixx="0.0001" ixy="0.0" ixz="0.0"
                iyy="0.0001" iyz="0.0" 
                izz="0.0001" />
	      </inertial>-->

        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <box size="0.001 0.001 0.001" />
            </geometry>
        </visual>
    </link>
    
    <link name="gps_antenna1">
    			<inertial>
      				<origin xyz="0.035 1.27350000005198E-07 0.084899" rpy="0 0 0" />
      				<mass value="0.28321" />
      				<inertia
        				ixx="0.000632544028273332"
        				ixy="-4.4923388876054E-20"
        				ixz="1.07422508310795E-20"
        				iyy="1.26609361466704E-09"
        				iyz="-0.000632544028277925"
        				izz="-1.26609361462022E-09" />
    			</inertial>
    			<visual>
      				<origin xyz="-0.035 0 0" rpy="1.5708 0 0" />
      				<geometry>
					<mesh filename="package://hratc2017_framework/models/fsr_husky_parts/gps_antenna.stl" />
      				</geometry>
      				<material name="White">
        				<color rgba="1 1 1 1"/>
      				</material>
    			</visual>
    			<collision>
      				<origin xyz="-0.035 0 0" rpy="1.5708 0 0" />
      				<geometry>
					<mesh filename="package://hratc2017_framework/models/fsr_husky_parts/gps_antenna.stl" />
     			 	</geometry>
    			</collision>
  		</link>
		
		<joint name="gps_antenna_joint1" type="fixed">
			<parent link="base_footprint"/>
			<child link="gps_antenna1"/>
			<origin xyz="0.0 0.0 1" rpy="0 0 0" /> 
		</joint>
		
		<gazebo reference="gps_antenna1">
        		<material>Gazebo/White</material>
        		<turnGravityOff>true</turnGravityOff>
		</gazebo>
		
		<gazebo>
			<plugin name="gazebo_ros_gps" filename="libgazebo_gps_plugin.so">
			  <updateRate>4.0</updateRate>
			  <bodyName>gps_antenna1</bodyName>
			  <topicName>/gps1/fix</topicName>
			  <velocityTopicName>/gps1/fix_velocity</velocityTopicName>
			  <offset>0 0 0</offset>
			  <drift>0 0 0</drift>
			  <gaussianNoise>0 0 0</gaussianNoise>
			</plugin>
    		</gazebo>
  
</robot>
